File: tests/benchmarks/resolution/fixtures/csharp/Service.cs
using System;

namespace Benchmark;

public class UserService
{
    private readonly IRepository _repo;

    public UserService(IRepository repo)
    {
        _repo = repo;
    }

    public User GetUser(string id)
    {
        return _repo.FindById(id);
    }

    public bool AddUser(User user)
    {
        if (!Validators.ValidateUser(user))
        {
            return false;
        }
        _repo.Save(user);
        return true;
    }

    public bool RemoveUser(string id)
    {
        var existing = _repo.FindById(id);
        if (existing == null) return false;
        return _repo.Delete(id);
    }
}

File: tests/benchmarks/resolution/fixtures/php/Service.php
<?php

require_once __DIR__ . '/Repository.php';
require_once __DIR__ . '/Validators.php';

class UserService
{
    private UserRepository $repo;

    public function __construct(UserRepository $repo)
    {
        $this->repo = $repo;
    }

    public function getUser(string $id): ?User
    {
        return $this->repo->findById($id);
    }

    public function addUser(User $user): bool
    {
        if (!Validators::validateUser($user)) {
            return false;
        }
        $this->repo->save($user);
        return true;
    }

    public function removeUser(string $id): bool
    {
        $existing = $this->repo->findById($id);
        if ($existing === null) {
            return false;
        }
        return $this->repo->delete($id);
    }
}
